### PR TITLE
feat: add page streaming for php

### PR DESCRIPTION
**Pull request by bshaffer**
This functionality now works for MapFields!

### CODE REVIEW

File: src/test/java/com/google/api/codegen/protoannotations/testdata/php_library_rest.baseline [hunk 1]
```diff
@@ -4365,20 +4396,26 @@ class LibraryServiceClientTest extends GeneratedTest
         $this->assertTrue($transport->isExhausted());
 
         // Mock response
-        $nextPageToken = 'nextPageToken-1530815211';
+        $nextPageToken = '';
+        $shelvesItem = new Shelf();
+        $shelves = ['shelves' => $shelvesItem];
```
Comment (bshaffer) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/php_library_rest.baseline:4401]: This should be `$shelves = [$shelvesItem];`. I'm investigating!

File: src/main/java/com/google/api/codegen/config/PageStreamingConfig.java [hunk 2]
```diff
@@ -246,8 +246,15 @@ static PageStreamingConfig createPageStreamingConfig(
...
       // TODO: Conform to design doc spec, once approved, for using non-standard paging fields
       //       (such as max_results for page_size)
-      if (language == TargetLanguage.JAVA && transportProtocol == TransportProtocol.HTTP) {
-        pageSizeField = methodModel.getInputField(pagingParams.getNameForMaxResults());
+      if (transportProtocol == TransportProtocol.HTTP) {
+        if (language == TargetLanguage.JAVA) {
+          pageSizeField = methodModel.getInputField(pagingParams.getNameForMaxResults());
+        } else if (language == TargetLanguage.PHP) {
+          FieldModel resourcesField = ProtoPageStreamingTransformer.getResourcesField(methodModel);
+          if (resourcesField != null && !resourcesField.isMap()) {
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/config/PageStreamingConfig.java:254]: Do we not need this in Java because Java *is* handling map responses?

**Review by vchudnov-g — COMMENTED** [review]
LGTM but have @vam-google TAL

File: src/main/java/com/google/api/codegen/config/PageStreamingConfig.java (hunk 2 above)
Comment (bshaffer) [reply src/main/java/com/google/api/codegen/config/PageStreamingConfig.java:254]: That is my understanding, yes!

**Review by vam-google — APPROVED** [review]
LGTM

File: src/main/resources/com/google/api/codegen/php/test.snip [hunk 3]
```diff
@@ -553,3 +557,14 @@
...
+@private mapFieldPageStreamingAsserts(view, initCode)
+    @join line : initCode.lines
+        @if line.lineType == "MapInitLine"
+            @join mapEntry : line.initEntries on ","
+                $this->assertEquals(
+                    $expectedResponse->{@view.resourcesFieldGetterName}()[{@mapEntry.key}], ${@view.resourcesVarName}[{@mapEntry.key}]);
+            @end
+        @end
+    @end
+@end
```
Comment (vam-google) [inline src/main/resources/com/google/api/codegen/php/test.snip:570]: minor: please add a new line

**Pull request merged**